src/jit/lower.cpp

```diff
@@ -3408,6 +3408,9 @@ bool Lowering::IndirsAreEquivalent(GenTreePtr candidate, GenTreePtr storeInd)
     if (pTreeA->OperGet() != pTreeB->OperGet())
         return false;
 
+    if (candidate->gtType != storeInd->gtType)
```

Comment:
Yikes, thanks for reminding me about ngen/crossgen, I could have done the diffs myself. This happens if `ref byte` or `byte*` are used, for example somewhere in mscorlib there's the following method:

```C#
private static unsafe void WriteNibble(ref byte* ptr, byte* endPtr, uint value)
{
    Contract.Assert(0 <= value && value < 16);
    Contract.Assert(ptr < endPtr);

    if (*ptr != 0)
        *ptr++ |= (byte)value; // this generates worse code with the current fix in place
    else
        *ptr = (byte)(value << 4);
}
```
No diffs are reported in mscorlib if the check I mentioned in a post above is used instead of the original check.